### PR TITLE
Fix state of services and internal classes when AppCenter started as disabled

### PR DESCRIPTION
**Pull request by MatkovIvan**
There is a bug when `AppCenter` was disabled in the previous session.
In this case, log manager, seder, channels, services are enabled after start.
This PR harmonized behavior when `AppCenter` disabled in current and previous session.

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m [hunk 1]
```diff
@@ -94,6 +96,11 @@ - (void)enqueueItem:(id<MSLog>)item withCompletion:(enqueueCompletionBlock)compl
     if (completion) {
       completion(success);
     }
+    
+    // Don't flush while disabled.
+    if (!self.enabled) {
```
Comment (bmourat) [inline AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m]: Do we need this when we have the same condition in flushQueue and startTimer?

File: AppCenter/AppCenter/MSAppCenter.m [hunk 2]
```diff
@@ -285,6 +280,13 @@ - (BOOL)startService:(Class)clazz andSendLog:(BOOL)sendLog {
 
     // Start service with log manager.
     [service startWithLogManager:self.logManager appSecret:self.appSecret];
+    
+    // Disable service if AppCenter is disabled.
+    if ([clazz isEnabled] && !self.isEnabled) {
+      self.enabledStateUpdating = YES;
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: @TroubleMakerBen @clpolet Do we really need this flag? I wonder if we synchronize `setEnabled` and get rid of this flag.

Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: @MatkovIvan Do you need to change this flag? You are not disabling the core services here.

**Comment by jaelim-ms** [comment]
@MatkovIvan Please resolve the conflict.

File: AppCenter/AppCenter/MSAppCenter.m (hunk 2 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/MSAppCenter.m]: @jaelim-ms This is caused by https://github.com/Microsoft/AppCenter-SDK-Apple/blob/9449f274db22f9d5c3777a3c749b4b1ff0afdf53/AppCenter/AppCenter/MSServiceAbstract.m#L88

Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: Gotcha

**Pull request merged**